flink-core/src/main/java/org/apache/flink/api/common/JobID.java

```diff
@@ -64,27 +67,48 @@ public JobID(long lowerPart, long upperPart) {
 	public JobID(byte[] bytes) {
 		super(bytes);
 	}
-	
+
 	// ------------------------------------------------------------------------
 	//  Static factory methods
 	// ------------------------------------------------------------------------
 
 	/**
 	 * Creates a new (statistically) random JobID.
-	 * 
+	 *
 	 * @return A new random JobID.
 	 */
 	public static JobID generate() {
 		return new JobID();
 	}
 
+	/**
+	 * Creates a new JobID based on the given seed. The JobIDs returned by two invocations of this
+	 * method with the same seed will be equal.
+	 *
+	 * @param seed the seed to base the generation of the JobID
+	 *
+	 * @return A new JobID based on the given seed.
+	 */
+	public static JobID fromSeed(String seed) {
```

Comment:
Just, imo, I'd move this method outside of `JobID` class (`StandaloneJobClusterEntryPoint` is probable host candidate):
 - this method is now less "generic" as before (although, before it was also specific);
 - it's computationally not cheap, and may be unintentionally used in other places.

Reply:
I have realized, that `JobID` is in fact marked as Flink public API. This makes the argument to move this method away from `JobID` even stronger.

Could you also add a test that passes a few hardcoded seed values and checks that the results of this calls return precomputed ids?

Reply:
I added a test as suggested and moved it to `StandalonJobClusterEntrypoint`. From an encapsulation point `JobID` would be the better place, and I don't worry about the performance implications (JobID creation is not critical), but it might be safer not to touch a public API at this point.